DENSE_utilities/DENSE_toolbox/AnalysisViewer.m

```diff
@@ -1538,44 +1538,55 @@ function deleteFcn()
 function [file, out] = exportMatFcn(obj,startpath)
 
     % check for startpath
-    if nargin < 2 || isempty(startpath) || ~exist(startpath,'dir')
+    if nargin < 2 || isempty(startpath)
         startpath = pwd;
     end
 
+    if ~exist(startpath, 'dir')
+        [origstartpath,~,ext] = fileparts(startpath);
+        if strcmpi(ext,'.mat')
+            file = startpath;
+            startpath = origstartpath;
```

Comment:
This probably need to be moved out of the if statement, otherwise this fails if I provide an extension other than .mat. Either that or we need to just check for a non-empty file extension but that would cause issues if the default path has a `.` in it. I would just move this line outside the if statement.